addons/sale_management/models/res_partner.py

```diff
@@ -0,0 +1,28 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import models, api, _
+
+
+class ResPartner(models.Model):
+    _inherit = 'res.partner'
+
+    @api.onchange('property_product_pricelist')
+    def _onchange_pricelist(self):
+        self.ensure_one()
+
+        open_order_count = self.env['sale.order'].search([
+            ('partner_id', 'child_of', self._origin.id),
+            ('state', 'in', ['draft', 'sent']),
+            ('pricelist_id', '!=', self.property_product_pricelist.id),
+        ], count=True)
+
+        if open_order_count > 0:
+            return {'warning': {
+                'title': _('Open Sale Orders'),
+                'message': _(
+                    'There is/are open sales order(s) for this partner. '
+                    'Please note that sales order pricelist will not be updated with this change. '
+                    'You should modify the sales order pricelist manually if needed.'
+                ),
```

Comment:
```suggestion
                'message': _(
                    'There are open quotations for this partner. '
                    'Please note that existing quotations (& sales orders) pricelists will not be updated with this change. '
                    'You should modify the sales order pricelist manually if needed.'
                ),
```

Draft and Sent SO are still considered Quotations, not true Sales Orders ;).
(It's a suggestion, feel free to further improve the message)

Reply:
This was changed in favor of an e-commerce specific message.